src/librbd/internal.cc

```diff
@@ -1524,23 +1526,57 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
 	return;
       }
 
-      Context *ctx = new C_CopyWrite(m_throttle, m_bl);
-      auto comp = io::AioCompletion::create(ctx);
-
-      // coordinate through AIO WQ to ensure lock is acquired if needed
-      m_dest->io_work_queue->aio_write(comp, m_offset, m_bl->length(),
-                                       std::move(*m_bl),
-                                       LIBRADOS_OP_FLAG_FADVISE_DONTNEED);
+      if (!m_sparse_size) {
+	m_sparse_size = (1 << m_dest->order);
+      }
+        
+      auto *end_op_ctx = new FunctionContext([this](int r) {
+	this->m_throttle->end_op(r);
```

Comment:
This instance will be automatically deleted by the time it leaves this function -- that means the ```FunctionContext``` here will have a pointer to a deleted object, which it will try to dereference and it will fail. You need to make a copy of the ```m_throttle``` pointer and pass the copy of the pointer into the lambda.